liquibase-core/src/main/java/liquibase/changelog/DatabaseChangeLog.java

```diff
@@ -570,6 +570,11 @@ public boolean include(String fileName,
             } else {
                 fileName = FilenameUtils.getFullPath(relativeBaseFileName) + fileName;
             }
+
+            // ADITO: normalize the file path to transform the path from x/foo/../bar to x/bar
+            // the normalize is there in newer versions: https://github.com/liquibase/liquibase/blob/master/liquibase-standard/src/main/java/liquibase/changelog/DatabaseChangeLog.java#L1011
+            fileName = Optional.ofNullable(FilenameUtils.normalizeNoEndSeparator(fileName)).orElse(fileName);
```

Comment:
Will this change be a problem for already existing systems?

Reply:
only if there are normal changelogs with includes that contain `..` somewhere in the path.

Reason for this change:
I had a RootAlias changelog and the NormalAlias changelog where my  RootAlias  changelog was referenced with
`../RootAlias/changelog.xml`.

When I executed Liquibase at RootAlias the filePath `.liquibase/RootAlias/changelog.xml` was written to the `DATABASECHANGELOG`  table.

When I used the new feature to execute Liquibase in the dist folder, and executed it on my NormalAlias, the path in the filePath would be `.liquibase/NormalAlias/../RootAlias/changelog.xml` and the change was executed a second time, if the changelog from RootAlias was already run.

Reply:
It seems, after a quick look at the corresponding repository, that the current "unmodularized" basic does not use ".." in the paths at all, so it should not be a problem here. Please correct me, if I am wrong here.
Otherwise the change should be OK.